CHANGELOG.md

```diff
@@ -0,0 +1,28 @@
+## 3.0.0 (2020-04-07)
+
+Complete rewrite of `CancelablePromise` by using `Proxy`.
+Now promises returned from `Promise` API such as `then` or `catch` can cancel the root promise and all promises created from this root promise:
+
+```javascript
+// CancelablePromise v2
+import CancelablePromise from 'cancelable-promise';
+
+const promise = new CancelablePromise((resolve) => setTimeout(resolve, 1))
+  .then(() => console.log('callback2 executed'))
+  .then(() => console.log('callback3 executed'));
+promise.cancel();
+// logs:
+// callback2 executed
+// callback3 executed
```

Comment:
v2 was not well working, to avoid these logs I should write:
```javascript
import CancelablePromise from 'cancelable-promise';

const promise = new CancelablePromise((resolve) => setTimeout(resolve, 1))
promise.then(() => console.log('callback2 executed'))
promise.then(() => console.log('callback3 executed'));
promise.cancel();
// no logs
```

Reply:
?

Reply:
I just commented the fact that v2 is executing callbacks in this case:
```javascript
const promise = new CancelablePromise((resolve) => setTimeout(resolve, 1))
  .then(() => console.log('callback2 executed'))
  .then(() => console.log('callback3 executed'));
promise.cancel();
// logs:
// callback2 executed
// callback3 executed
```

Reply:
You mean, CancelablePromise doesn't cancel ? Or it's a edge case with setTimeout(..., 1) ?